Topic 0/src/point3/Factory.java

```diff
@@ -0,0 +1,16 @@
+package point3;
+
+public class Factory {
+	
+	public Hospital getHospital(String dimension){
+		Director director= null;
+		if (dimension.equals("small"))
+			director = new Director(new RuralHospital()); 
+		else 
+			if (dimension.equals("big"))
+				director = new Director(new CityHospital()); 
+		director.biuld();
```

Comment:
[Medium] In case that *dimesion* is not any of allowed values (small, big), then *director* will be <code>null</code> and you will get an *NullPointerException*.
That's why it is important to handle that situation. You can throw your own exception to do that saying: 'You must select a valid Hospital type' for example.